pandas/indexes/category.py

```diff
@@ -459,21 +459,13 @@ def _convert_list_indexer(self, keyarr, kind=None):
 
         return None
 
-    def take(self, indexer, axis=0, allow_fill=True, fill_value=None):
-        """
-        For internal compatibility with numpy arrays.
-
-        # filling must always be None/nan here
-        # but is passed thru internally
-        assert isnull(fill_value)
-
-        See also
-        --------
-        numpy.ndarray.take
-        """
-
-        indexer = com._ensure_platform_int(indexer)
-        taken = self.codes.take(indexer)
+    @Appender(_index_shared_docs['take'])
+    def take(self, indices, axis=0, allow_fill=True, fill_value=None):
+        indices = com._ensure_platform_int(indices)
```

Comment:
If we move it to ``assert_take_fillable``, it will be executed duplicatelly when ``take`` has additional logic`` like below. No need to care?

- https://github.com/pydata/pandas/pull/12676/files#diff-b56ec8d28d045a5a83219fe4d5e4dabfR266